### PR TITLE
[WIP] Added missing Registry query endpoints

### PR DESCRIPTION
**Pull request by drgomesp**
#302  WIP

## Changes

- Item 1
- Item 2

## Checklist

- [ ] Unit tests
- [ ] Documentation

## References *(optional)*

Fixes 
Connects 

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: internal/highway/crud_registry.go [hunk 1]
```diff
@@ -1,57 +1,122 @@
...
+// @Router /registry/who_is[GET]
+func (s *HighwayServer) ParamsHTTP(c *gin.Context) {
+	params, err := s.cosmos.QueryParams()
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+
+	c.JSON(http.StatusOK, params)
+}
```
Comment (joshLong145) [inline internal/highway/crud_registry.go]: I dont think we need to expose query params over http.

File: internal/highway/crud_registry.go [hunk 2]
```diff
@@ -1,57 +1,122 @@
...
+// @Router /registry/who_is[GET]
+func (s *HighwayServer) WhoIsHTTP(c *gin.Context) {
+	names, err := s.cosmos.QueryAllNames()
+	if err != nil {
+		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		return
+	}
+
+	c.JSON(http.StatusOK, names)
+}
```
Comment (joshLong145) [inline internal/highway/crud_registry.go]: This should not be exposed over http, we do not wish to allow the user to do an exhaustive query of this data model.

File: internal/highway/crud_registry.go [hunk 3]
```diff
@@ -1,57 +1,122 @@
...
 		}
 
-		// Save Registration Session
-		options, err := s.webauthn.SaveRegistrationSession(c.Request, c.Writer, username, s.cosmos.AccountName())
+		options, err := s.webauthn.SaveRegistrationSession(
+			c.Request,
+			c.Writer,
+			username,
+			s.cosmos.AccountName(),
+		)
```
Comment (joshLong145) [inline internal/highway/crud_registry.go:94]: remove formatting as this goes against the go fmt convention to my understanding

File: internal/highway/main.go [hunk 4]
```diff
@@ -226,6 +226,9 @@ func setupAPI(ctx context.Context, s *HighwayServer) error {
 	s.router.POST("/v1/object/create", s.CreateObjectHTTP)
 	s.router.POST("/v1/object/update", s.UpdateObjectHTTP)
 	s.router.POST("/v1/object/deactivate", s.DeactivateObjectlHTTP)
+	s.router.GET("/v1/registry/params", s.ParamsHTTP)
```
Comment (joshLong145) [inline internal/highway/main.go]: remove routing def for query params

File: internal/highway/proto/v1/highway.proto [hunk 5]
```diff
@@ -82,4 +82,18 @@ service Highway {
...
+  rpc QueryParams(sonrio.sonr.registry.QueryParamsRequest) returns (sonrio.sonr.registry.QueryParamsResponse) {}
+
+  // QueryWhoIs
+  //
+  //
+  rpc QueryWhoIs(sonrio.sonr.registry.QueryWhoIsRequest) returns (sonrio.sonr.registry.QueryWhoIsResponse) {}
+
+  // QueryWhoIsAll
+  //
+  rpc QueryWhoIsAll(sonrio.sonr.registry.QueryWhoIsAllRequest) returns (sonrio.sonr.registry.QueryAllWhoIsResponse) {}
```
Comment (joshLong145) [inline internal/highway/proto/v1/highway.proto]: Please remove new proto definitions as this is not something we wish to expose over grpc

File: pkg/client/cosmos.go [hunk 6]
```diff
@@ -149,28 +149,42 @@ func (cc *Cosmos) NameExists(name string) bool {
...
+		golog.Errorf("Error querying all names: %s", err.Error())
+		return nil, err
+	}
+
+	params := queryResp.GetParams()
+
+	return &params, nil
+}
+
+// QueryAllNames ...
```
Comment (joshLong145) [inline pkg/client/cosmos.go]: Please remove

**Review by joshLong145 — CHANGES_REQUESTED** [review]
See review comments

File: internal/highway/crud_registry.go (hunk 3 above)
Comment (joshLong145) [reply internal/highway/crud_registry.go:94]: Sorry, this is valid go formatting. ... duh, resolving this

File: internal/highway/crud_registry.go (hunk 1 above)
Comment (drgomesp) [reply internal/highway/crud_registry.go]: I was basically exposing the methods defined in the query.proto (https://github.com/sonr-io/sonr/blob/dev/proto/registry/query.proto) for the Registry. If they are not supposed to be exposed, which ones are considered missing and would need to be added?

Comment (drgomesp) [reply internal/highway/crud_registry.go]: Can I follow https://www.notion.so/sonr/ADR-1-Sonr-DID-Method-Specification-0677c6f389794787bf40772bb00f6f8f#ae710980849747ecb699dc83265436b4 as reference?

It specifies:

```
// Find a DIDDocument by ID
(GET): /v1/registry/query/whois/:did
- DID:string

// Find a DIDDocument with the matching controller
(GET): /v1/registry/query/whois/controller/:did
- DID:string

// Find a DIDDocument with the matching alias
(GET): /v1/registry/query/whois/alias/:name
- Name:string
```

Comment (joshLong145) [reply internal/highway/crud_registry.go]: Let me follow up on this but for now ill resolve the comment. thanks!

Comment (drgomesp) [reply internal/highway/crud_registry.go]: Done

File: internal/highway/main.go (hunk 4 above)
Comment (drgomesp) [reply internal/highway/main.go]: done

File: internal/highway/proto/v1/highway.proto (hunk 5 above)
Comment (drgomesp) [reply internal/highway/proto/v1/highway.proto]: done

File: pkg/client/cosmos.go (hunk 6 above)
Comment (drgomesp) [reply pkg/client/cosmos.go]: done

**Comment by drgomesp** [comment]
Closed in favour of #325

**Pull request closed without merging**